### PR TITLE
luci-app-advanced-reboot: (Create xiaomi-ax9000.json)

### PR DESCRIPTION
**Pull request by tiagofreire-pt**

### CODE REVIEW

**Comment by tiagofreire-pt** [comment]
Wiki: https://openwrt.org/inbox/toh/xiaomi/ax9000#experimental_firmware

**Comment by stangri** [comment]
Thank you for this again!

Could you please confirm/clarify a few things:

1. What is the actual device name in `/tmp/sysinfo/board_name`? For some other models which have been supported longer, their names changed between OpenWrt releases, so multiple names are listed in json file for backwards compatibility. For the new device, one name should be sufficient.
2. Do you actually have `/dev/rootfs` and `/dev/rootfs_1` partitions? For most other devices partitions are either `/dev/mtdX` or `/dev/mmcblk0pX`.
3. Have you tested this by putting this file into `/usr/share/advanced-reboot/devices/` on the live router?

**Comment by tiagofreire-pt** [comment]
```
-----------------------------------------------------
 OpenWrt SNAPSHOT, r21394-1d82c6b4e3
 -----------------------------------------------------
root@AX9000:~# cat /tmp/sysinfo/board_name
xiaomi,ax9000
root@AX9000:~# cat /proc/mtd
dev:    size   erasesize  name
mtd0: 00100000 00020000 "0:sbl1"
mtd1: 00100000 00020000 "0:mibib"
mtd2: 00080000 00020000 "0:bootconfig"
mtd3: 00080000 00020000 "0:bootconfig1"
mtd4: 00300000 00020000 "0:qsee"
mtd5: 00300000 00020000 "0:qsee_1"
mtd6: 00080000 00020000 "0:devcfg"
mtd7: 00080000 00020000 "0:devcfg_1"
mtd8: 00080000 00020000 "0:apdp"
mtd9: 00080000 00020000 "0:apdp_1"
mtd10: 00080000 00020000 "0:rpm"
mtd11: 00080000 00020000 "0:rpm_1"
mtd12: 00080000 00020000 "0:cdt"
mtd13: 00080000 00020000 "0:cdt_1"
mtd14: 00080000 00020000 "0:appsblenv"
mtd15: 00100000 00020000 "0:appsbl"
mtd16: 00100000 00020000 "0:appsbl_1"
mtd17: 00080000 00020000 "0:art"
mtd18: 00080000 00020000 "bdata"
mtd19: 00100000 00020000 "pstore"
mtd20: 03800000 00020000 "rootfs"
mtd21: 03800000 00020000 "rootfs_1"
mtd22: 00100000 00020000 "cfg_bak"
mtd23: 07d80000 00020000 "overlay"
```

File: luci-app-advanced-reboot/root/usr/share/advanced-reboot/devices/xiaomi-ax9000.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+	"vendorName": "Xiaomi",
+	"deviceName": "AX9000",
+	"boardNames": [ "ax9000", "xiaomi,ax9000" ],
```
Comment (stangri) [inline luci-app-advanced-reboot/root/usr/share/advanced-reboot/devices/xiaomi-ax9000.json]:
```suggestion
	"boardNames": [ "xiaomi,ax9000" ],
```

File: luci-app-advanced-reboot/root/usr/share/advanced-reboot/devices/xiaomi-ax9000.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	"partition1MTD": "rootfs",
```
Comment (stangri) [inline luci-app-advanced-reboot/root/usr/share/advanced-reboot/devices/xiaomi-ax9000.json]:
```suggestion
	"partition1MTD": "mtd20",
```

File: luci-app-advanced-reboot/root/usr/share/advanced-reboot/devices/xiaomi-ax9000.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	"partition2MTD": "rootfs_1",
```
Comment (stangri) [inline luci-app-advanced-reboot/root/usr/share/advanced-reboot/devices/xiaomi-ax9000.json]:
```suggestion
	"partition2MTD": "mtd21",
```

**Comment by tiagofreire-pt** [comment]
https://imgur.com/QtjHDBd

**Comment by tiagofreire-pt** [comment]
I have a dual firmware boot, with still the OEM firmware on ```rootfs``` and the experimental openwrt firmware on ```rootfs_1```, compiled by me for the target IPQ8072, from the robimarko's repo.

On the next few days, I'll flash the rootfs with openwrt, as was today the first day that this "international" hardware variant got support: https://forum.openwrt.org/t/openwrt-support-for-xiaomi-ax9000/98908/909

**Comment by stangri** [comment]
> I have a dual firmware boot, with still the OEM firmware on `rootfs` and the experimental openwrt firmware on `rootfs_1`, compiled by me for the target IPQ8072, from the robimarko's repo.

It should read the kernel version from OpenWrt at least then, if the mtd and offset values are correct.

**Review by stangri — DISMISSED** [review]

**Comment by tiagofreire-pt** [comment]
> > I have a dual firmware boot, with still the OEM firmware on `rootfs` and the experimental openwrt firmware on `rootfs_1`, compiled by me for the target IPQ8072, from the robimarko's repo.
> 
> It should read the kernel version from OpenWrt at least then, if the mtd and offset values are correct.

The offset value seems to be correct for the robimarko's firmware, as the OEM one is based on an ancient openwrt version (18.0x).

https://imgur.com/7yJMPPl

**Review by stangri — COMMENTED** [review]
change partitions

**Comment by tiagofreire-pt** [comment]
With partitions changed:

image

**Comment by stangri** [comment]
Great, thank you again for this, it should be available from my private repo as `luci-app-advanced-reboot 1.0.1-4` momentarily.

PS. Sorry, should have asked you to bump a version in the Makefile too. ;)

**Pull request merged**